### ISSUE TITLE
Various class methods should be class properties

### ISSUE THREAD

**Issue by 1ec5** [mapbox/mapbox-gl-native#6778]
Various methods like `+[MGLStyle streetsStyleURL]`, `+[MGLAccountManager accessToken]`/`+[MGLAccountManager setAccessToken:]`, and `+[MGLOfflineStorage sharedOfflineStorage]` should be class properties with refined Swift names (#5178). Swift code would be able to say:

``` swift
MGLOfflineStorage.shared.addPack(…
```

Once #6386 is fixed, Swift code would be able to say:

``` objc
mapView.style = .streets
```

This change will require anyone compiling or building or using the SDK in another project to use Xcode 8.0 or above. It would also require a major version bump if we aren’t able to keep the old methods around as deprecated aliases.

/cc @friedbunny @frederoni

**Comment by friedbunny** [comment]
Agreed, this would be a good change.

**Comment by 1ec5** [comment]
Fixed in #11674 on the release-boba branch for iOS map SDK v4.0.0 and macOS map SDK v0.7.0.
